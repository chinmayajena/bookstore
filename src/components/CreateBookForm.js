import React from "react";
import { reduxForm, Field } from "redux-form";

export let CreateBook = props => {
  const { handleSubmit } = props;
  return (
    <div className="row">
      <form onSubmit={handleSubmit} className="form col-md-6 offset-md-3">
        <div className="form-group">
          <div className="control">
            <Field
              name="name"
              component={renderField}
              type="text"
              label="Book Name"
              className="form-control"
            />
          </div>
        </div>

        <div className="form-group">
          <div className="control">
            <Field
              name="author"
              component={renderField}
              type="text"
              label="Author Name"
            />
          </div>
        </div>

        <div className="form-group">
          <div className="control">
            <Field
              name="description"
              component={renderField}
              type="text"
              label="Description"
            />
          </div>
        </div>

        <div className="form-group">
          <div className="control">
            <label className="label">Category</label>
            <div className="select">
              <Field
                className="form-control"
                name="category"
                component="select"
              >
                <option value="">Select a category...</option>
                <option value="fiction">fiction</option>
                <option value="non-fiction">non-fiction</option>
                <option value="Drama">Drama</option>
              </Field>
            </div>
          </div>
        </div>

        <div className="form-check">
          <div className="control">
            <label className="form-check-label">
              <Field
                name="isBookmarked"
                id="bookmarkbook"
                component="input"
                type="checkbox"
                className="form-check-input"
              />
              Bookmark ?
            </label>
          </div>
        </div>

        <div className="form-group">
          <div className="control">
            <label className="label">Content</label>
            <Field
              className="form-control"
              name="content"
              component="textarea"
            />
          </div>
        </div>

        <div className="form-group">
          <div className="control">
            <button type="submit" className="btn btn-primary">
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

export const validate = val => {
  const errors = {};
  if (!val.bookName) {
    console.log("Book Name is required");
    errors.bookName = "Required";
  }
  if (!val.authorName) {
    console.log("Author Name is required");
    errors.authorName = "Required";
  }
  return errors;
};

export const renderField = ({
  input,
  label,
  type,
  meta: { touched, error, warning }
}) => (
  <div>
    <div className="control">
      <label className="field">{label}</label>
      <input
        className="form-control"
        {...input}
        placeholder={label}
        type={type}
      />
      {touched &&
        ((error && <span>{error}</span>) ||
          (warning && <span>{warning}</span>))}
    </div>
  </div>
);

let CreateBookForm = reduxForm({
  form: "createBook",
  enableReinitialize: true,
  keepDirtyOnReinitialize: true,
  updateUnregisteredFields: true
})(CreateBook);

export default CreateBookForm;
