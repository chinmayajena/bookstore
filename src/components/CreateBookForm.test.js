import React from "react";
import { shallow } from "enzyme";
import { CreateBook } from "./CreateBookForm";
import configureStore from "redux-mock-store";

describe("<CreateBookForm />", () => {
  const initialState = {};
  const mockStore = configureStore();
  let store, component;

  beforeEach(() => {
    store = mockStore(initialState);
    component = shallow(<CreateBook store={store} />);
  });

  it("renders the component", () => {
    expect(component.debug()).toMatchSnapshot();
    const wrapper = component.find(".form");
    expect(wrapper.length).toBe(1);
  });
});
