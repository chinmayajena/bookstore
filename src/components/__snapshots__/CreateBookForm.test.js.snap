// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateBookForm /> renders the component 1`] = `
"<div className=\\"row\\">
  <form onSubmit={[undefined]} className=\\"form col-md-6 offset-md-3\\">
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <Field name=\\"name\\" component={[Function: renderField]} type=\\"text\\" label=\\"Book Name\\" className=\\"form-control\\" />
      </div>
    </div>
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <Field name=\\"author\\" component={[Function: renderField]} type=\\"text\\" label=\\"Author Name\\" />
      </div>
    </div>
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <Field name=\\"description\\" component={[Function: renderField]} type=\\"text\\" label=\\"Description\\" />
      </div>
    </div>
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <label className=\\"label\\">
          Category
        </label>
        <div className=\\"select\\">
          <Field className=\\"form-control\\" name=\\"category\\" component=\\"select\\">
            <option value=\\"\\">
              Select a category...
            </option>
            <option value=\\"fiction\\">
              fiction
            </option>
            <option value=\\"non-fiction\\">
              non-fiction
            </option>
            <option value=\\"Drama\\">
              Drama
            </option>
          </Field>
        </div>
      </div>
    </div>
    <div className=\\"form-check\\">
      <div className=\\"control\\">
        <label className=\\"form-check-label\\">
          <Field name=\\"isBookmarked\\" id=\\"bookmarkbook\\" component=\\"input\\" type=\\"checkbox\\" className=\\"form-check-input\\" />
          Bookmark ?
        </label>
      </div>
    </div>
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <label className=\\"label\\">
          Content
        </label>
        <Field className=\\"form-control\\" name=\\"content\\" component=\\"textarea\\" />
      </div>
    </div>
    <div className=\\"form-group\\">
      <div className=\\"control\\">
        <button type=\\"submit\\" className=\\"btn btn-primary\\">
          Submit
        </button>
      </div>
    </div>
  </form>
</div>"
`;
